File: app/src/main/java/com/goel/attendancetracker/organisations/OrganisationsAdapter.java
                                                                                                                                            package com.goel.attendancetracker.organisations;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.shapes.Shape;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.goel.attendancetracker.MainActivity;
import com.goel.attendancetracker.R;

import java.util.ArrayList;

public class OrganisationsAdapter extends RecyclerView.Adapter<OrganisationsAdapter.viewHolder> {

    ArrayList<OrganisationsModel> organisationList;
    Context context;
    Drawable progress;
    private OnItemClickListener clickListener;

    public OrganisationsAdapter(ArrayList<OrganisationsModel> organisationList, Context context) {
        this.organisationList = organisationList;
        this.context = context;
    }

    @NonNull
    @Override
    public viewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.organisation_card, parent, false);
        return new viewHolder(view, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull viewHolder holder, int position) {
        OrganisationsModel model = organisationList.get(position);
        int percentage = model.getOrganisationAttendancePercentage();
        int requiredPercentage = model.getRequiredAttendance();
        if (percentage>=requiredPercentage)
            progress = context.getDrawable(R.drawable.attendance_progress);
        else if (percentage > requiredPercentage*0.75f)
            progress = context.getDrawable(R.drawable.attendance_progress_low);
        else
            progress = context.getDrawable(R.drawable.attendance_progress_danger);
        holder.attendancePercentage.setText(percentage + "%");
        holder.attendanceProgressBar.setProgressDrawable(progress);
        holder.attendanceProgressBar.setProgress(percentage);
        holder.requiredAttendanceBar.setProgress(requiredPercentage);
        holder.organisationName.setText(model.getOrganisationName());
        holder.editIcon.setImageResource(R.drawable.icon_edit);
        holder.deleteIcon.setImageResource(R.drawable.icon_delete);
    }

    @Override
    public int getItemCount() {
        return organisationList.size();
    }

    // ITEM CLICK LISTENER
    public interface OnItemClickListener{
        void onItemClick(int position);
        void onDeleteClick(int position);
        void onEditClick(int position);
    }
    public void setOnItemClickListener(OnItemClickListener listener){ clickListener = listener; }

    //VIEW HOLDER CLASS
    public static class viewHolder extends RecyclerView.ViewHolder{

        ProgressBar attendanceProgressBar;
        ProgressBar requiredAttendanceBar;
        TextView attendancePercentage;
        TextView organisationName;
        ImageView editIcon;
        ImageView deleteIcon;

        public viewHolder(@NonNull View itemView, OnItemClickListener listener) {
            super(itemView);
            attendanceProgressBar = itemView.findViewById(R.id.organisation_attendance_progress_bar);
            requiredAttendanceBar = itemView.findViewById(R.id.required_attendance_progress);
            attendancePercentage = itemView.findViewById(R.id.organisation_attendance_percentage);
            organisationName = itemView.findViewById(R.id.organisation_name);
            editIcon = itemView.findViewById(R.id.edit_organisation_icon);
            deleteIcon = itemView.findViewById(R.id.delete_organisation_icon);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null)
                    {
                        int position = getAdapterPosition();
                        if (position !=RecyclerView.NO_POSITION)
                            listener.onItemClick(position);
                    }
                }
            });

            deleteIcon.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null)
                    {
                        int position = getAdapterPosition();
                        if (position !=RecyclerView.NO_POSITION)
                            listener.onDeleteClick(position);
                    }
                }
            });

            editIcon.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null)
                    {
                        int position = getAdapterPosition();
                        if (position !=RecyclerView.NO_POSITION)
                            listener.onEditClick(position);
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/goel/attendancetracker/adapters/OrganisationsAdapter.kt
package com.goel.attendancetracker.adapters

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.goel.attendancetracker.R
import com.goel.attendancetracker.models.OrganisationsModel
import com.goel.attendancetracker.adapters.OrganisationsAdapter.OrganisationViewHolder
import java.util.*

class OrganisationsAdapter(
    private var organisationList: ArrayList<OrganisationsModel>,
    var context: Context
) : RecyclerView.Adapter<OrganisationViewHolder>() {
    private var progress: Drawable? = null
    private val clickListener = context as OnItemClickListener
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OrganisationViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.organisation_card, parent, false)
        return OrganisationViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: OrganisationViewHolder, position: Int) {
        val model = organisationList[position]
        val percentage = model.attendance
        val requiredPercentage = model.target
        progress = when {
            percentage >= requiredPercentage -> ContextCompat.getDrawable(
                context,
                R.drawable.attendance_progress
            )
            percentage > requiredPercentage * 0.75f -> ContextCompat.getDrawable(
                context, R.drawable.attendance_progress_low
            )
            else -> ContextCompat.getDrawable(
                context, R.drawable.attendance_progress_danger
            )
        }
        holder.attendancePercentage.text = "$percentage%"
        holder.attendanceProgressBar.progressDrawable = progress
        holder.attendanceProgressBar.progress = 0
        holder.attendanceProgressBar.progress = percentage
        holder.requiredAttendanceBar.progress = requiredPercentage
        holder.organisationName.text = model.name
        setClickListeners(holder, position)
    }

    override fun getItemCount(): Int {
        return organisationList.size
    }

    private fun setClickListeners(
        holder: OrganisationViewHolder,
        position: Int
    ) {
        if (position != RecyclerView.NO_POSITION) {
            holder.itemView.setOnClickListener { clickListener.onItemClick(position) }
            holder.editIcon.setOnClickListener { clickListener.onEditClick(position) }
            holder.deleteIcon.setOnClickListener { clickListener.onDeleteClick(position) }
        }
    }

    // ITEM CLICK LISTENER
    interface OnItemClickListener {
        fun onItemClick(position: Int)
        fun onDeleteClick(position: Int)
        fun onEditClick(position: Int)
    }

    //VIEW HOLDER CLASS
    class OrganisationViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var attendanceProgressBar: ProgressBar = itemView.findViewById(R.id.organisation_attendance_progress_bar)
        var requiredAttendanceBar: ProgressBar = itemView.findViewById(R.id.required_attendance_progress)
        var attendancePercentage: TextView = itemView.findViewById(R.id.organisation_attendance_percentage)
        var organisationName: TextView = itemView.findViewById(R.id.organisation_name)
        var editIcon: ImageView = itemView.findViewById(R.id.edit_organisation_icon)
        var deleteIcon: ImageView = itemView.findViewById(R.id.delete_organisation_icon)
    }
}